observer/probers/tls/tls.go

```diff
@@ -0,0 +1,165 @@
+package probers
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/crypto/ocsp"
+	"io"
+	"net/http"
+	"time"
+)
+
+var StatusCode int
+
+const (
+	Success = iota
+	UnexpectedRoot
+	UnexpectedResponse
+	UnexpectedRootAndResponse
+	UnknownOCSPCheck
+	FailedOCSPCheck
+	FailedTLSConnection
+	Unknown
+)
```

Comment:
If you're going to use `iota` you'll need to provide a function for translating between integer and string so that what ends up in the underlying metric label is a string. That being said, I think we really only have a few possible values we actually want making it into the label.

```go
// Could be good or bad depending on what we configured the check to look for.
OCSPGood
OCSPRevoked
OCSPUnknown
// The check failed internally.
IssuerVerifyFailed
InternalError
```